### ISSUE TITLE
Python test `test_import_file_disk_full` fail with `object has no attribute 'get_manifest'` on Windows

### ISSUE THREAD

**Issue by FirelightFlagboy** [Scille/parsec-cloud#4259]
On the job <https://github.com/Scille/parsec-cloud/actions/runs/4477127429/jobs/7868278164>,

The test `tests/core/gui/test_files.py::test_import_file_disk_full` fail with:

```
ERROR    parsec.core.mountpoint.winfsp_operations:winfsp_operations.py:175 2023-03-21 09:11:58 [error    ] Unhandled exception in winfsp mountpoint [parsec.core.mountpoint.winfsp_operations] mountpoint=P:\ operation=get_security_by_name path=/AutoRun.inf workspace_id=EntryID(e6127f9b-3718-4791-9a69-b4ca46beda9c)
Traceback (most recent call last):
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\winfsp_operations.py", line 128, in get_path_and_translate_error
    yield path
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\winfsp_operations.py", line 255, in wrapper
    return func.__get__(self)(arg, *args, **kwargs)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\winfsp_operations.py", line 309, in get_security_by_name
    self.fs_access.entry_info(parsec_file_name)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 145, in entry_info
    return self._run(self.workspace_fs.transactions.entry_info, path)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 110, in _run
    return _run_from_thread_to_trio(self._trio_token, afn, self.DEADLOCK_TIMEOUT, *args)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 90, in _run_from_thread_to_trio
    return queue.get().unwrap()
  File "C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs\parsec-cloud-_mDprxAt-py3.9\lib\site-packages\outcome\_impl.py", line 138, in unwrap
    raise captured_error
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 67, in unprotected_afn
    return await afn(*args)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 252, in entry_info
    manifest, confinement_point = await self._get_manifest_from_path(path)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 137, in _get_manifest_from_path
    entry_id, confined = await self._entry_id_from_path(path)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 108, in _entry_id_from_path
    manifest = await self._load_manifest(entry_id)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 91, in _load_manifest
    async with self._load_and_lock_manifest(entry_id) as manifest:
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\contextlib.py", line 181, in __aenter__
    return await self.gen.__anext__()
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 75, in _load_and_lock_manifest
    local_manifest = await self.local_storage.get_manifest(entry_id)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\storage\workspace_storage.py", line 89, in wrapper
    return await func(self, *args, **kwargs)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\storage\workspace_storage.py", line 295, in get_manifest
    return await self.rs_instance.get_manifest(entry_id)
AttributeError: 'NoneType' object has no attribute 'get_manifest'
```

This is likekly be caused by the `canary` that we put on `WorkspaceStorage::run`:

https://github.com/Scille/parsec-cloud/blob/d48f651d68294dcd34f7a423a887b2b121dff383/parsec/core/fs/storage/workspace_storage.py#L173-L178

The question is now: Why a thread was accessing the `WorkpsaceStorage` that have been closed by the context manager ?

**Comment by FirelightFlagboy** [comment]
https://github.com/Scille/parsec-cloud/actions/runs/4593723344/jobs/8111935625

```
 Traceback (most recent call last):
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\winfsp_operations.py", line 128, in get_path_and_translate_error
    yield path
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\winfsp_operations.py", line 255, in wrapper
    return func.__get__(self)(arg, *args, **kwargs)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\winfsp_operations.py", line 309, in get_security_by_name
    self.fs_access.entry_info(parsec_file_name)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 145, in entry_info
    return self._run(self.workspace_fs.transactions.entry_info, path)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 110, in _run
    return _run_from_thread_to_trio(self._trio_token, afn, self.DEADLOCK_TIMEOUT, *args)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 90, in _run_from_thread_to_trio
    return queue.get().unwrap()
  File "C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs\parsec-cloud-_mDprxAt-py3.9\lib\site-packages\outcome\_impl.py", line 138, in unwrap
    raise captured_error
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\mountpoint\thread_fs_access.py", line 67, in unprotected_afn
    return await afn(*args)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 252, in entry_info
    manifest, confinement_point = await self._get_manifest_from_path(path)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 138, in _get_manifest_from_path
    manifest = await self._load_manifest(entry_id)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 91, in _load_manifest
    async with self._load_and_lock_manifest(entry_id) as manifest:
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\contextlib.py", line 181, in __aenter__
    return await self.gen.__anext__()
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\workspacefs\entry_transactions.py", line 75, in _load_and_lock_manifest
    local_manifest = await self.local_storage.get_manifest(entry_id)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\storage\workspace_storage.py", line 89, in wrapper
    return await func(self, *args, **kwargs)
  File "D:\a\parsec-cloud\parsec-cloud\parsec\core\fs\storage\workspace_storage.py", line 295, in get_manifest
    return await self.rs_instance.get_manifest(entry_id)
AttributeError: 'NoneType' object has no attribute 'get_manifest'
```

**Comment by FirelightFlagboy** [comment]
This test was removed, closing the issue

**Comment by FirelightFlagboy** [comment]
Fixed by #5528 on releases/2.16
